### ISSUE TITLE
Getting core UX right

### ISSUE THREAD

**Issue by andreis** [lavab/web#766]
Our UI looks fantastic and many things work smoothly, but overall it's still rather clunky to use. As soon as the plugin-based architecture (PBA?) is done I want to focus on getting core UX right.

- [ ] Optimistic caching https://github.com/lavab/web/issues/649
  - Example https://github.com/rubenv/angular-optimistic-cache
- [ ] Compose interface - replace TextAngular with something decent, ideally Markdown-based
  - I'm strongly in favour of HTML editor by default, with Markdown split view as an alternative view. Plaintext mode shouldn't be needed, as Markdown will allow editing the actual text (we could use something like [this](https://domchristie.github.io/to-markdown/) to convert back to Md). It just depends on how committed we want to be to Markdown.
  - [ ] Decrease the size of the line spacing, it's way too big currently.
- [ ] Drafts
  - [ ] Auto-save
  - [ ] Drafts "folder"
- [ ] Displaying emails
  - I think there is still work to be done here. Most of the time I'm not happy with how emails look in the right pane. Sorry that I can't really provide a better answer here.

![screen shot 2015-05-22 at 3 10 41 am](https://cloud.githubusercontent.com/assets/1249760/7762139/3c06bcc6-0030-11e5-94d2-c92b0b8d5949.png)

- [ ] (Optional-ish) Quick in-line replies
  - When in doubt, Do What Mailbox Does™ (click pic for video)
  - This will be easier to do if we have our own editor methinks
  - Should be implemented as a plugin (Sergey :) )

[![screen shot 2015-05-22 at 3 23 23 am](https://cloud.githubusercontent.com/assets/1249760/7762259/0050d638-0032-11e5-9395-4f0bd6b15f1b.png)](https://www.youtube.com/watch?v=9lJ2a0D759s)

- [ ] Cross-browser testing.
  - Many subtle issues can appear on different browsers/OSes. Side comment, I recently installed Fedora and Ubuntu to do some ad-hoc testing there.
  - We have a free browserstack subscription, since we're an OSS project. @flixi has details

Feel free to make suggestions and additions to this list. I really believe we need to get the core user experience right before moving on to more involved features.

**Comment by vLooz** [comment]
totally with you @andreis 
there are still some element, that need fine-tuning. Let me provide you guys with mocks how the UI should look eventually. I think most times we are quite close, but the details make the difference.

**Comment by kalicki** [comment]
![3c06bcc6 0030 11e5 94d2 c92b0b8d5949 png 1552x985](https://cloud.githubusercontent.com/assets/3674885/7770289/43deaca0-0063-11e5-9677-9c20255e37d6.png)

Hey guys, I have a question, I find it very confusing that part, as it does not have information of each item (e.g. tooltip in mousehover), you don't think the same?

IMHO

**Comment by piggyslasher** [comment]
Yes, mouseovers are coming for the email actions buttons #781. And I was an early adopter of GhostLab. Its much better than BrowserSync but I think they've gone premium now. My app still works fine though.

http://vanamco.com/ghostlab/

We use LiveReload in Gulp right now and it fits our needs _for now_ :) But I run this through GhostLab so even my scrolling is synchronized across devices.

**Comment by piggyslasher** [comment]
+1 for the embedded editor in the emails. Emails are going to be displayed much nicer now. Update tomorrow to develop. On .com soon after hopefully.

**Comment by flixi** [comment]
Attachment:
+ support for binary attachments, so .pdf and other filetypes sent to lavaboom users
+ support for stuff sent non-encrypted

**Comment by piggyslasher** [comment]
Binary attachments work. I've tried with a small mp3 as well as a zip file with an image inside. Both work.

**Comment by vLooz** [comment]
Great job everyone! :-)

**Comment by let4be** [comment]
Binary attachments work only when sent to a secure destination: i.e. to another lavaboom or to external email with attached public key...
Binary attachments to unsecure destinations are blocked and can't be implemented atm, please see: https://github.com/lavab/web/issues/692
